subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/NormalizedPathFingerprintCompareStrategy.java

```diff
@@ -106,17 +107,30 @@ protected boolean doVisitChangesSince(
         for (Entry<String, FileSystemLocationFingerprint> entry : currentFingerprints.entrySet()) {
             String currentAbsolutePath = entry.getKey();
             FileSystemLocationFingerprint currentFingerprint = entry.getValue();
-            List<FilePathWithType> previousFilesForFingerprint = missingPreviousFiles.get(currentFingerprint);
-            if (previousFilesForFingerprint.isEmpty()) {
+            boolean wasAccountedFor = accountFor(missingPreviousFiles, currentAbsolutePath, currentFingerprint);
+            if (!wasAccountedFor) {
```

Comment:
If we use "accounted for" as a concept here, then I think we really should keep using it consistently, i.e. keep it for `unaccoutnedForPreviousFiles` as well.